job-descriptions/software-engineer-code-exercise.md

```diff
@@ -0,0 +1,60 @@
+# Code exercise
+
+We ask candidates to complete a 2-hour asynchronous coding exercise so we can evaluate you on certain skills that we think are relevant to being successful at Sourcegraph.
+
+Why? Read the [FAQ](#FAQ).
+
+## TypeScript code exercise
+
+We designed this exercise to measure your understanding of callbacks and asyncronous execution.
+
+- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time; we just schedule an email.
+- At your chosen date and time, we will send you a zipped TypeScript project that contains instructions and some unit tests.
+- TypeScript is a superset of JavaScript, so if you are comfortable with JavaScript and have experience reading any typed language, then you should be fine. We recommend reading through the [TypeScript docs](https://www.typescriptlang.org/docs/handbook/basic-types.html) if you aren't already familiar with TypeScript.
+- You should have Node.js (>=8.16.0) installed on your computer.
+- You can look up documentation on the internet while you are coding.
+- You may use open-source libraries, but most candidates don't find it necessary.
+- After two hours, you will email us your solution as a zip file and we will get back to you on next steps within 2 business days.
+
+If we decide to move forward, we will schedule a 30-minute followup call to discuss your code.
+
+## Go code exercise
+
+We designed this exercise to measure your understanding of HTTP requests, concurrency, and error handling.
+
+- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time that works for you; there are no scheduling constraints.
+- At your chosen date and time, we will send you detailed instructions. You will be building a simple API server in Go that calls a JSON endpoint of another service and exposes a JSON endpoint of its own.
+- You should have a recent version of Go installed on your computer.
```

Comment:
Should we be specific?

```suggestion
- You should have a recent version of Go (≥ 1.11) installed on your computer.
```

Reply:
@tsenart @keegancsmith What is the right minimum version to document here?

Reply:
I picked 1.11 in my suggestion to ensure it supports modules (they did backport some of that to 1.10, but things get rough). We probably shouldn't suggest anything _earlier_ than that, but it'd be perfectly rational to insist on 1.12 or 1.13.

Reply:
1.11 is a good minimum due to gomod. :+1: